src/lib/staticVersioner.js

```diff
@@ -0,0 +1,101 @@
+const fs = require('fs').promises;
+const path = require('path');
+
+async function versionStaticAssets(sitePaths, staticAssets, version) {
+    let staticDir = sitePaths.staticDir;
+    let versionDocsDir = sitePaths.versionedDocs;
+
+    for (const staticType of staticAssets) {
+        console.info(`Versioning ${staticType} asset files...`);
+        let staticTypePath = path.join(staticDir, staticType);
+        let staticTypeNextPath = path.join(staticTypePath, "next");
+
+        /*
+            If the next directory does not exist for a static asset type, then it is the first time it is versioned. The
+            catch block will create the next directory and update the relative paths in the docs and versioned_docs
+            directory. The then block will just update the path in the version_docs directory.
+         */
+        await fs.access(staticTypeNextPath)
+            .then(async () => {
+                let baseVersionedDocsPath = path.join(versionDocsDir, `version-${version}`);
+                let relativeLinkPattern = new RegExp(`../../${staticType}/next/`, 'gm');
+                let replacementText = `../${staticType}/${version}/`;
+                await updateRelativePaths(baseVersionedDocsPath, relativeLinkPattern, replacementText);
+            })
+            .catch(async () => {
+                await copyDirectory(staticTypePath, staticTypeNextPath);
+                await removeFilesInDirectory(staticTypePath, "next");
+
+                let relativeLinkPattern = new RegExp(`../${staticType}/`, 'gm');
+                let replacementText = `../../${staticType}/next/`;
+                await updateRelativePaths(sitePaths.docs, relativeLinkPattern, replacementText);
+
+                replacementText = `../${staticType}/${version}/`;
+                let baseVersionedDocsPath = path.join(versionDocsDir, `version-${version}`);
+                await updateRelativePaths(baseVersionedDocsPath, relativeLinkPattern, replacementText);
+            })
+        let staticTypeVersionPath = path.join(staticTypePath, version);
+        await copyDirectory(staticTypeNextPath, staticTypeVersionPath)
```

Comment:
Let's move these into the 2 then-catch blocks to guarantee their execution in appropriate order.